tests/Doctrine/Tests/DBAL/Schema/MySqlSchemaManagerTest.php

```diff
@@ -15,7 +15,7 @@ class MySqlSchemaManagerTest extends \PHPUnit\Framework\TestCase
      */
     private $manager;
 
-    /** @var Connection */
+    /** @var Connection|\PHPUnit_Framework_MockObject_MockObject */
```

Comment:
As far as I understand, `phpstan/phpstan-phpunit` should take care of mock types.

Reply:
See also:

> `createMock()`, `getMockForAbstractClass()` and `getMockFromWsdl()` methods return an intersection type (see the [detailed explanation of intersection types](https://medium.com/@ondrejmirtes/union-types-vs-intersection-types-fd44a8eacbb)) of the mock object and the mocked class so that both methods from the mock object (like `expects`) and from the mocked class are available on the object.

Doesn't it mean that PHPStan should infer proper mock types without hints? Or there's a conflict between an explicit type annotation and PHPStan-inferred? I.e. without the doc-block, PHPStan would understand that the object is a connection _and_ a mock but the doc-block states that it's only a connection.